kubernetes/resource_kubernetes_annotations_test.go

```diff
@@ -74,6 +79,148 @@ func TestAccKubernetesAnnotations_basic(t *testing.T) {
+			{
+				Config: testAccKubernetesAnnotations_template_empty(name),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "api_version", "batch/v1"),
+					resource.TestCheckResourceAttr(resourceName, "kind", "CronJob"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.name", name),
+					resource.TestCheckResourceAttr(resourceName, "annotations.%", "0"),
+					resource.TestCheckResourceAttr(resourceName, "template_annotations.%", "0"),
+					resource.TestCheckResourceAttr(resourceName, "field_manager", "tftest"),
+				),
+			},
+			{
+				Config: testAccKubernetesAnnotations_template_basic(name),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "api_version", "batch/v1"),
+					resource.TestCheckResourceAttr(resourceName, "kind", "CronJob"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.name", name),
+					resource.TestCheckResourceAttr(resourceName, "annotations.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "annotations.test1", "one"),
+					resource.TestCheckResourceAttr(resourceName, "template_annotations.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "template_annotations.test2", "two"),
+					resource.TestCheckResourceAttr(resourceName, "field_manager", "tftest"),
+				),
+			},
+			{
+				Config: testAccKubernetesAnnotations_template_modified(name),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "api_version", "batch/v1"),
+					resource.TestCheckResourceAttr(resourceName, "kind", "CronJob"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.name", name),
+					resource.TestCheckResourceAttr(resourceName, "annotations.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "annotations.test1", "one"),
+					resource.TestCheckResourceAttr(resourceName, "annotations.test2", "two"),
+					resource.TestCheckResourceAttr(resourceName, "template_annotations.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "template_annotations.test3", "three"),
+					resource.TestCheckResourceAttr(resourceName, "template_annotations.test4", "four"),
+					resource.TestCheckResourceAttr(resourceName, "field_manager", "tftest"),
+				),
+			},
+			{
+				Config: testAccKubernetesAnnotations_template_empty(name),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "api_version", "batch/v1"),
+					resource.TestCheckResourceAttr(resourceName, "kind", "CronJob"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.name", name),
+					resource.TestCheckResourceAttr(resourceName, "annotations.%", "0"),
+					resource.TestCheckResourceAttr(resourceName, "template_annotations.%", "0"),
+					resource.TestCheckResourceAttr(resourceName, "field_manager", "tftest"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccKubernetesAnnotations_template_deployment(t *testing.T) {
```

Comment:
Let's add tests for the cases when only `annotations` or `template_annotations` is set to cover the above panic scenario.